packages/component-library/src/components/plmg-radio-button/plmg-radio-button.tsx

```diff
@@ -0,0 +1,134 @@
+import { Component, h, Prop, Watch } from '@stencil/core';
+import {
+  PlmgRadioButtonSize,
+  isPlmgRadioButtonSize,
+} from './plmg-radio-button.types';
+
+@Component({
+  tag: 'plmg-radio-button',
+  styleUrl: 'plmg-radio-button.scss',
+  shadow: false,
+})
+export class RadioButton {
+  private inputElement: HTMLInputElement;
+
+  /**
+   * Define radio button's size.
+   *
+   * Allowed values:
+   *   - medium
+   *   - large
+   *
+   * Default: medium
+   */
+  @Prop() size: PlmgRadioButtonSize = 'medium';
+  @Watch('size')
+  validateSize(newValue: string) {
+    if (!isPlmgRadioButtonSize(newValue))
+      throw new Error('size: must be a valid value');
+  }
+
+  /**
+   * Define radio button's value'
+   */
+  @Prop() value: string;
+  @Watch('value')
+  validateValue(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('value: required');
+    if (newValue && typeof newValue !== 'string')
+      throw new Error('value must be a string');
+  }
+
+  /**
+   * Callback to provide validity of radio input to
+   * radio button group
+   */
+  @Prop() isValid: (valid: boolean) => void;
```

Comment:
My comment there is bad 😅 

It's not really meant for the external users of plumage (I'd prefer if we could somehow not expose this prop to them). It only makes sense with the radio-button-group. We always pass in this same value for the `isValid` prop in the radio-button-group component:

```
(inputValid: boolean) => {
    if (this.isValid !== inputValid) {
         this.isValid = inputValid;
    }
 }
```

We use this just to bubble up the validation pass/fail from the individual radio button back up to the radio button group, so we can show the error-message (or not). Otherwise the radio-button-group has no idea if the form validation failed on required radio buttons.

Does that make sense?

Reply:
Gotcha, thanks!